### DISCUSSION TITLE
Allow custom Attributes in column relationships

### DISCUSSION THREAD

**Discussion by joostvanhoof** [filamentphp/filament#13270] [Ideas]
If I have a `User` model, with `first_name` and `last_name` as database columns, and this custom attribute on the `User` model:

```php
protected function name(): Attribute
{
    return Attribute::make(
        get: fn ($value, $attributes) => $attributes['first_name'].' '.$attributes['last_name']
    );
}
```

Then these will work:
```php
TextColumn::make('users.first_name')
TextColumn::make('users.last_name')
```

But this will not:
```php
TextColumn::make('users.name')
```

Would be nice if custom attributes also work here.

**Comment by leandrocfe** [comment]
You could use a virtual column 

`$table->string('full_name')->virtualAs('concat(first_name, \' \', last_name)');`

TextColumn::make('full_name')

**Reply by atorscho** [reply]
this requires a new migration to update the DB which is not always possible, unfortunately

**Reply by krekas** [reply]
Why not? Why can't you run one additional migration?
